### PR TITLE
✨Move tkg vmware-system-* annotations to vmi/cvmi status

### PR DESCRIPTION
**Pull request by sreyasn**
**What does this PR do, and why is it needed?**

One of the goals w/ this is to make Metadata watches more effective in v1a2 by not having big yamls in the ObjectMeta.

This change moves the
- tkg vmware-system-* annotations to a v1a2 status field
- adds conversions for this status field to convert to annotations for v1a1.
- v1a1 annotations when up-converted are copied over to the v1a2 status field and removed from v1a2 annotations if already present

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes # N/A

**Are there any special notes for your reviewer**:
Testing TKC deployments and scaled up to check vmware-system annotations are fetched from the v1a1 virtualmachineimage.

**Please add a release note if necessary**:

```
N/A
```

### CODE REVIEW

**Comment by akutz** [comment]
Let's make sure we also engage TKG internally to jump-start their move to v1a2.

File: api/v1alpha1/virtualmachineimage_conversion.go [hunk 1]
```diff
@@ -308,6 +345,11 @@ func (src *VirtualMachineImage) ConvertTo(dstRaw conversion.Hub) error {
 		return err
 	}
 
+	if err := convert_v1alpha1_VirtualMachineImageAnnotations_To_v1alpha2_VirtualMachineImage_VMwareSystemProperties(
```
Comment (bryanv) [inline api/v1alpha1/virtualmachineimage_conversion.go:348]: We want to remove the annotations that we're moved to the status. One of the goals w/ this is to make MD-only watches more effective by not have the big TKG yaml blog in the ObjectMeta.

File: webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go [hunk 2]
```diff
@@ -349,8 +349,8 @@ func (v validator) validateNetworkInterfaceSpec(
 		networkIfCRName = fmt.Sprintf("%s-%s", vmName, interfaceSpec.Name)
 	}
 
-	for _, msg := range validation.NameIsDNSLabel(networkIfCRName, false) {
-		allErrs = append(allErrs, field.Invalid(interfacePath.Child("name"), interfaceSpec.Name, msg))
+	for _, msg := range validation.NameIsDNSSubdomain(networkIfCRName, false) {
```
Comment (bryanv) [inline webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go:352]: This looks fine but I misspoke yesterday - let's do this as a separate MR since it isn't related to the primary change in the MR.

File: api/v1alpha1/virtualmachineimage_conversion.go (hunk 1 above)
Comment (sreyasn) [reply api/v1alpha1/virtualmachineimage_conversion.go:362]: fixed it.. PTAL when you get a chance

**Review by bryanv — APPROVED** [review]
